### DISCUSSION TITLE
Download hour and day sections

### DISCUSSION THREAD

**Discussion by Mr-CostlessClaw** [blakeblackshear/frigate#5247] [Ideas]
I came across the need to download more than just clips associated with events. To pull off days of recordings. The idea to be able to download the individual hours from the recording section as well as individual days. It could just be 24 hour clips as a 1 day file.
Though, when people are using this for security and they need to show someone or law-enforcement proof. Downloading clips are inconvenient. Not to say anything bad about Frigate and your work.

**Comment by NickM-27** [comment]
this can already be done with the API, there is just not a button in the UI. This is currently complicated due to the fact that combining clips into a single file requires a lot of space and most people don't have enough RAM for a full day. There is already an existing feature request for this

**Reply by Mr-CostlessClaw** [reply]
Thanks. I am trying to learn the HTTP API. The system in question is on a LAN only. I only hae access to it physically. I never got my system back up and running. I got to set a camara up to learn this.
Though surprised that the files goes to RAM first. 
Though I am not trying to prod. I know the UI shows hour increments, at the same time I left a thumbdrive to copy the data from the media file frigate uses.

**Reply by NickM-27** [reply]
When you watch in the UI it does not go to RAM, however when you try to download 1 hour as a single mp4 file it must be created somewhere because the recordings are stored as 10 second segments internally.

**Reply by Mr-CostlessClaw** [reply]
That makes since to the hundreds of files in the media file. I was wondering if I was going to have to stich that together to make one video. Apparently so. Got to do what I got to do. Thanks for all that. It was very informative.

**Reply by NickM-27** [reply]
Not sure what you mean. This is very easy to do with ffmpeg or VLC

**Reply by NickM-27** [reply]
Just need to use the m3u8 API

**Reply by Mr-CostlessClaw** [reply]
I have been looking at this.
https://docs.frigate.video/integrations/api/#get-vodyear-monthdayhourcameramasterm3u8
Though my personal system doesn't show what it has. 
So I am working on it atm to learn this. 

And I was going to use VLC or handbreak to merge the files.

This computer has no networking. When downloading alternative plug ins. I have to go and retrieve it. Bring it home and work on it.

And thanks again for all this. It means a lot.

**Reply by blakeblackshear** [reply]
There is no need to do anything fancy to merge files. Ffmpeg will do it all for you. See here: https://github.com/blakeblackshear/frigate/issues/5040#issuecomment-1381861765

**Reply by Mr-CostlessClaw** [reply]
This kinda looks like this command
/vod/< year>-< month>/< day>/< hour>/< camera>/master.m3u8#(got working in VLC)
I am unsure about this this part.  ffmpeg -i ..... -c copy clip.mp4
If this is ran in the config file or Extra Parameters:
I had figured out that first command. though I was trying to figure out this command,
/vod/< camera>/start/< start-timestamp>/end/< end-timestamp>/index.m3u8#(Still working on)
I could not figure out how the timestamps were labeled for VLC.
That would allow me to put. 2023-01-05/19-2023-01-08/19.

Though I did read over the comment and the whole thread. And this is well above my head.
I am still thankful for all the help. I get to learn a lot about this software. and would like to contribute eventually. This is as far as I have gotten today.
         PS. I have no clue why the commands and text disappeared. :/
